proposals/relaxed-simd/Overview.md

```diff
@@ -37,6 +37,77 @@ Example of some instructions we would like to add:
 - Relaxed Swizzle (implementation defined out of bounds behavior)
 - Relaxed Rounding Q-format Multiplication (optional saturation)
 
+### Consistency
+
+This proposal introduces non-deterministic instructions - given the same
+inputs, two calls to the same instruction can return different results. For
+example:
+
+```wast
+(module
+  (func (param v128 v128 v128)
+    (f32x4.qfma (local.get 0) (local.get 1) (local.get 2))   ;; (1)
+    ;; some other computation
+    (f32x4.qfma (local.get 0) (local.get 1) (local.get 2)))) ;; (2)
+```
```

Comment:
It seems like most users would tend to want consistent results, so it would seem that toolchains would want to use the immediates in most cases.

Also, once we start looking at the binary format, I expect it won't be feasible to have an optional immediate field, because there's no place to indicate whether the field is present or not. One way to approach this would be to have two versions of these opcodes, one with an immediate and one without, however that seems fairly redundant.

What would you think of making the immediate non-optional?

Reply:
I think this example caused some confusion, I wrote this example to illustrate the case where we don't have a `fpenv` construct, and what problems it would have. I agree with you that the instructions should *always* take an immediate, having two versions is redundant.

Reply:
Ah, yes, I was confused just as you say.